File: backend/src/dtos/Orphanage.kt
package com.lorenzoog.happy.dtos

import com.lorenzoog.happy.models.Orphanage
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class CreateOrphanageData(
    val name: String,
    val latitude: String,
    val longitude: String,
    val about: String,
    val instructions: String,
    @SerialName("open_on_weekends") val openOnWeekends: Boolean = false,
    @SerialName("opening_hours") val openingHours: String,
)

@Serializable
data class OrphanageResponse(
    val id: Long,
    val name: String,
    val latitude: String,
    val longitude: String,
    val about: String,
    val instructions: String,
    @SerialName("open_on_weekends") val openOnWeekends: Boolean = false,
    @SerialName("opening_hours") val openingHours: String,

    val images: List<String>
)

fun Orphanage.asResponseDto(images: List<String>) = OrphanageResponse(
    id,
    name,
    latitude.toString(),
    longitude.toString(),
    about,
    instructions,
    openOnWeekends,
    openingHours,
    images
)

File: mobile/lib/models/orphanage.dart
class Orphanage {
  final num id;
  final String name;
  final num latitude;
  final num longitude;
  final String about;
  final String instructions;
  final String openingHours;
  final bool openOnWeekends;
  final List<String> images;

  const Orphanage({
    this.id,
    this.name,
    this.latitude,
    this.longitude,
    this.about,
    this.instructions,
    this.openingHours,
    this.openOnWeekends,
    this.images,
  });

  Orphanage.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        latitude = num.parse(json['latitude']),
        longitude = num.parse(json['longitude']),
        about = json['about'],
        instructions = json['instructions'],
        openingHours = json['opening_hours'],
        openOnWeekends = json['open_on_weekends'],
        images = (json['images'] as List<dynamic>)
            .map((object) => object.toString())
            .toList();
}
